gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageImpl.java

```diff
@@ -1965,10 +1965,11 @@ private StorageObject getObject(StorageResourceId resourceId)
   private boolean canIgnoreExceptionForEmptyObject(
       IOException exceptionOnCreate, StorageResourceId resourceId, CreateObjectOptions options)
       throws IOException {
-    // TODO(user): Maybe also add 409 and even 412 errors if they pop up in this use case.
+    // TODO(user): Maybe also add 409 errors if they pop up in this use case.
     // 500 ISE and 503 Service Unavailable tend to be raised when spamming GCS with create requests:
     if (errorExtractor.rateLimited(exceptionOnCreate)
-        || errorExtractor.internalServerError(exceptionOnCreate)) {
+        || errorExtractor.internalServerError(exceptionOnCreate)
+        || errorExtractor.preconditionNotMet(exceptionOnCreate)) {
```

Comment:
Probably we need ignore this only for directories?

Reply:
Done